src/gui/guiFormSpecMenu.cpp

```diff
@@ -2065,6 +2263,19 @@ void GUIFormSpecMenu::regenerateGui(v2u32 screensize)
 		}
 	}
 
+	/* Copy of the "real_coordinates" element for before the form size. */
+	mydata.real_coordinates = false;
+	for (; i< elements.size(); i++) {
```

Comment:
This is an optional formspec element. Use a copy of `i` in case this element is not provided.

Reply:
`parseElement` only parses elements after `size`, `position`, and `anchor`.  So, if it is only parsed there, then the formsize will use the old coordinates since this tag only applies to things after it.

As for a copy of `i`, this isn't necessary because it checks if the element is `size` or `invsize` after checking for `real_coordinates`, and it isn't incremented if it is.  This check can't be removed even with a copy of `i` because it might parse a `real_coordinates` after the size, giving an incorrect result.

Reply:
Sorry, my bad. The loop breaks before `i` is incremented - exactly as it should be. Since the order of these initial elements is fixed, it might be safer to `break;` in any case. In fact, `!elements[i].empty()` allows incorrectly written formspecs, such as `formspec_version[1]]]]]real_coordinates[true]`.

Reply:
So, are you saying I should make it like https://github.com/v-rob/minetest/blob/formspec-real-coordinates/src/gui/guiFormSpecMenu.cpp#L2298-#L2300 instead of what I'm using now (`!elements[i].empty()`)?

Reply:
Yes, exactly. That could look like this:
```C++
mydata.real_coordinates = false;
for (; i < elements.size(); i++) {
	std::vector<std::string> parts = split(elements[i], '[');
	std::string name = trim(parts[0]);
	if (name != "real_coordinates"parts.size() != 2)
		break; // Invalid format

	mydata.real_coordinates = is_yes(trim(parts[1]));
}
```